### PR TITLE
feat(element-registry): add ElementRegistry#find method

### LINKED ISSUE #261: Add ElementRegistry#find method
We offer a number of collection-like utilities on the `ElementRegistry` already.

What is missing is the ability to find the first element that matches a certain condition.

Right now the workaround is:

```
var element = elementRegistry.filter(condition)[0];
```

...which will always iterate over the complete list of elements.

__Tasks__

* [ ] Provide `ElementRegistry#find` method

### PR DESCRIPTION
**Pull request by stdevi**
Closes #261

### CODE REVIEW

File: lib/core/ElementRegistry.js [hunk 1]
```diff
@@ -137,6 +137,29 @@ ElementRegistry.prototype.filter = function(fn) {
...
+ */
+ElementRegistry.prototype.find = function(fn) {
+
+  var map = this._elements;
+
+  for (var i = 0; i < Object.keys(map).length; i++) {
+    var id = Object.keys(map)[i];
+    var container = map[id],
+        element = container.element,
+        gfx = container.gfx;
```
Comment (pinussilvestrus) [inline lib/core/ElementRegistry.js:155]: Better

```js
var id = Object.keys(map)[i],
       container = map[id],
       element = container.element,
       gfx = container.gfx;
```

File: lib/core/ElementRegistry.js [hunk 2]
```diff
@@ -137,6 +137,29 @@ ElementRegistry.prototype.filter = function(fn) {
...
+ *
+ * @param {Function} fn
+ *
+ * @return {djs.model.Base}
+ */
+ElementRegistry.prototype.find = function(fn) {
+
+  var map = this._elements;
+
+  for (var i = 0; i < Object.keys(map).length; i++) {
```
Comment (pinussilvestrus) [inline lib/core/ElementRegistry.js:151]: what about storing the keys in a separate variable, so you don't have to execute it twice?

```js
var keys = Object.keys(map)
```

File: test/spec/core/ElementRegistrySpec.js [hunk 3]
```diff
@@ -206,6 +206,59 @@ describe('ElementRegistry', function() {
 
   });
 
+  describe('find', function() {
```
Comment (pinussilvestrus) [inline test/spec/core/ElementRegistrySpec.js:209]: Make sure you have _two new lines_ between any test-group (describe) and any test (it).

File: lib/core/ElementRegistry.js (hunk 2 above)
Comment (stdevi) [reply lib/core/ElementRegistry.js:151]: Agree, I will do that.

**Comment by pinussilvestrus** [comment]
Thanks for your contribution! I especially like that you consider what the author of the issue (https://github.com/bpmn-io/diagram-js/issues/261) has said

> ...which will always iterate over the complete list of elements.

and prevent this by early return when one match was found. Can you also update your commit message to fit our guidelines?

```js
feat(ElementRegistry): add find method

Closes #261 
```

File: test/spec/core/ElementRegistrySpec.js [hunk 4]
```diff
@@ -206,6 +206,62 @@ describe('ElementRegistry', function() {
 
   });
 
+
+  describe('find', function() {
```
Comment (pinussilvestrus) [inline test/spec/core/ElementRegistrySpec.js:210]: Should we maybe also add a test case, what happens, when no match was found?

File: lib/core/ElementRegistry.js [hunk 5]
```diff
@@ -137,6 +137,30 @@ ElementRegistry.prototype.filter = function(fn) {
   return filtered;
 };
 
+/**
+ * Return the first elements that satisfies the provided testing function.
```
Comment (pinussilvestrus) [inline lib/core/ElementRegistry.js:141]: "the first element" (no plural)

File: test/spec/core/ElementRegistrySpec.js (hunk 4 above)
Comment (stdevi) [reply test/spec/core/ElementRegistrySpec.js:210]: Sure, I've added one additional test case.

**Comment by pinussilvestrus** [comment]
thanks a lot! I merged it via cb45b735dd7ed24c28b9138ed4dd3f36baa84c11

**Pull request closed without merging**